avocado/utils/network/hosts.py

```diff
@@ -122,7 +122,15 @@ def __init__(self, host, username, port=22, key=None, password=None):
         self.username = username
         self.key = key
         self.password = password
-        self.remote_session = self._connect()
+        self.remote_session = None
```

Comment:
Before this patch, self.remote_session it was initialized during object initialization. __enter__ is not called when object is being initialized. This change here is not necessary.

Reply:
Oh! TIL. Thank you